trojsten/contests/templatetags/archive_parts.py

```diff
@@ -1,15 +1,21 @@
+from collections import OrderedDict
 from django import template
 
 from trojsten.contests.models import Competition
 
-from ..helpers import get_rounds_by_year
+from ..helpers import get_rounds_by_year, slice_tag_list
 
 register = template.Library()
 
 
+def slice_tags(round_tag):
+    return round_tag[0], slice_tag_list(list(round_tag[1]))
+
+
 @register.inclusion_tag('trojsten/contests/parts/round_list.html')
 def show_round_list(user, competition):
     all_rounds = get_rounds_by_year(user, competition)
+    all_rounds = OrderedDict((key, list(map(slice_tags, v))) for key, v in all_rounds.items())
```

Comment:
Optional: Vo vseobecnosti nie som fanusikom strkania 2 roznych veci do tej istej premennej - toto sa da aj spravit aj inline.

Ale uz je to ovela krajsie, vdaka!

Reply:
Nazvy premennych by tu mohli popisovat tie objekty, ktore su key a value miesto nic-nehvoriacich `key` a `v`.